.github/workflows/release.yml

```diff
@@ -77,51 +77,47 @@ jobs:
-            ' .github/workflows/_buildpacks-prepare-release.yml
+        run: |
+          yq -i '
+            (.jobs[].steps[] |
+              select(.uses == "${{ github.repository }}/.github/actions/install-languages-cli@main") |
+              .uses
+            ) = "${{ github.repository }}/.github/actions/install-languages-cli@v${{ steps.metadata.outputs.version }}"
+          ' .github/workflows/_buildpacks-prepare-release.yml
 
       - name: Update workflows/_buildpacks-release.yml
-        uses: mikefarah/yq@v4.34.2
-        with:
-          cmd: |
-            yq -i '
-              (.jobs[].steps[] |
-                select(.uses == "${{ github.repository }}/.github/actions/install-languages-cli@main") |
-                .uses
-              ) = "${{ github.repository }}/.github/actions/install-languages-cli@v${{ steps.metadata.outputs.version }}"
-            ' .github/workflows/_buildpacks-release.yml
-
-      - name: Add changes to release branch
         run: |
-          git checkout -b v${{ steps.metadata.outputs.version }}
+          yq -i '
+            (.jobs[].steps[] |
+              select(.uses == "${{ github.repository }}/.github/actions/install-languages-cli@main") |
+              .uses
+            ) = "${{ github.repository }}/.github/actions/install-languages-cli@v${{ steps.metadata.outputs.version }}"
+          ' .github/workflows/_buildpacks-release.yml
+
+      - name: Create a release tag with required changes
+        id: release-tag
+        run: |
           git add .
           git commit -m "Release v${{ steps.metadata.outputs.version }}"
-          git push -u origin v${{ steps.metadata.outputs.version }}
+          git tag v${{ steps.metadata.outputs.version }}
+          git push origin refs/tags/v${{ steps.metadata.outputs.version }}
           git tag -f latest
           git push -f origin refs/tags/latest
+          echo "sha=$(git rev-list -n 1 v${{ steps.metadata.outputs.version }})" >> $GITHUB_OUTPUT
 
       - name: Create release
         uses: softprops/action-gh-release@v0.1.15
         env:
           GITHUB_TOKEN: ${{ steps.generate-token.outputs.app_token }}
         with:
-          target_commitish: v${{ steps.metadata.outputs.version }}
+          target_commitish: ${{ steps.release-tag.outputs.sha }}
           tag_name: v${{ steps.metadata.outputs.version }}
           files: ${{ steps.package-binary.outputs.path }}
           fail_on_unmatched_files: true
-          body: "Full Changelog: https://github.com/${{ github.repository }}/compare/v${{ steps.previous-version.outputs.value }}..v${{ steps.metadata.outputs.version }}"
+          body: "Full Changelog: https://github.com/${{ github.repository }}/compare/${{ steps.previous-version.outputs.value == '0.0.0' && 'main' || format('v{0}', steps.previous-version.outputs.value) }}..v${{ steps.metadata.outputs.version }}"
```

Comment:
If the `0.0.0` fallback is removed higher up in the workflow (xref the other thread on this review), then this fallback will need to be removed too.